mathml/relations/html5-tree/html-or-foreign-element-interfaces.tentative.html

```diff
@@ -0,0 +1,139 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8" />
+    <title>MathML 'HTMLOrForeignElement` Mixin Tests</title>
+    <link
+      rel="help"
+      href="https://mathml-refresh.github.io/mathml-core/#dom-and-javascript"
+    />
+    <style>
+        mi { background-color: red;  }
+        :focus { background-color: rgb(0, 255, 0); }
+      }
+    </style>
+    <meta name="assert" content="MathMLElements incorporate a functional HTMLOrForeignElement interface" />
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+
+  </head>
+  <body tabindex="-1">
+    <span tabindex="-1">This tests the presence and functionality of features of `HTMLOrForeignElement` (currently `HTMLOrSVGElement`)</span>
+    <math
+      tabindex="-1"
+      xmlns="http://www.w3.org/1998/Math/MathML"
+      display="block"
+    >
+      <mi>E</mi>
+    </math>
+  </body>
+  <script>
+
+    // spot check the functionality of several interfaces
+    let el = document.querySelector("mi");
+    let mathEl = document.querySelector("math");
+
+    // this really belongs in 
+    // https://github.com/web-platform-tests/wpt/blob/master/html/dom/elements/global-attributes/dataset.html
+    // it is here tentatively
+    test(function() {
+      var mathml = document.createElementNS(
+        "http://www.w3.org/1998/Math/MathML",
+        "math"
+      );
+      assert_true(mathml.dataset instanceof DOMStringMap);
```

Comment:
can't you just do mathEl.dataset and/or el.dataset instead of creating a new root?

Reply:
I can, but see the note - ultimately we would want to move that particular one to  https://github.com/web-platform-tests/wpt/blob/master/html/dom/elements/global-attributes/dataset.html

but until we get that agreed to then I think this is just making it complicated